From b7eb0ac5f4185d74e8a7e270ead056df7eef8258 Mon Sep 17 00:00:00 2001
From: Maksymilian Goryszewski <mgoryszewski@internships.antmicro.com>
Date: Fri, 24 Apr 2026 08:31:13 +0200
Subject: [PATCH 1/1] Change memory region used by malloc()

---
 lib/libc/common/source/stdlib/malloc.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/lib/libc/common/source/stdlib/malloc.c b/lib/libc/common/source/stdlib/malloc.c
index 74193fbdce4..6eeca587054 100644
--- a/lib/libc/common/source/stdlib/malloc.c
+++ b/lib/libc/common/source/stdlib/malloc.c
@@ -4,6 +4,7 @@
  * SPDX-License-Identifier: Apache-2.0
  */

+#include <zephyr/linker/devicetree_regions.h>
 #include <stdlib.h>
 #include <zephyr/kernel.h>
 #include <zephyr/init.h>
@@ -82,7 +83,11 @@ K_APPMEM_PARTITION_DEFINE(z_malloc_partition);
 #   define HEAP_SIZE	ROUND_UP(CONFIG_COMMON_LIBC_MALLOC_ARENA_SIZE, HEAP_ALIGN)
 #   define HEAP_BASE	POINTER_TO_UINT(malloc_arena)

+#if defined(CONFIG_BOARD_STM32H747I_DISCO)
+static Z_GENERIC_SECTION(LINKER_DT_NODE_REGION_NAME(DT_NODELABEL(sdram2))) unsigned char __aligned(HEAP_ALIGN) malloc_arena[HEAP_SIZE];
+#else
 static POOL_SECTION unsigned char __aligned(HEAP_ALIGN) malloc_arena[HEAP_SIZE];
+#endif

 #  else /* CONFIG_COMMON_LIBC_MALLOC_ARENA_SIZE > 0 */

--
2.39.5
